src/iterable.cr

```diff
@@ -33,4 +33,54 @@ module Iterable
   def each_with_object(obj)
     each.with_object(obj)
   end
+
+  # :nodoc:
+  # Overrides Comparable.==
+  # This version works with more objects because it doesn't rely on <=>.
+  def ==(other: self)
+    equals?(other) { |x, y| x == y }
+  end
+
+  # Iterates over both Iterables comparing each value against other's values.
+  # returns true if all elements are equal in both Iterables.
+  # return false otherwise.
+  # ```
+  # ary = [1, 2, 3]
+  # ary == [1, 2, 3] # => true
+  # ary == [2, 3] # => false
+  # ```
+  def ==(other : Iterable)
+    equals?(other) { |x, y| x == y }
+  end
+
+  # :nodoc:
+  def ==(other)
+    false
+  end
+
+  def equals?(other)
+    other_iter = other.each
+    each do |item|
+      other_item = other_iter.next
+      return false if other_item.is_a?(Iterator::Stop)
+      return false unless yield(item, other_item)
+    end
+    return true if other_iter.next.is_a?(Iterator::Stop)
+    false
+  end
+
+  module FixedSizeCompare(T)
```

Comment:
Comparable requires <=>.  Doing so would restrict the types of things that could be compared.

This uses == for greater compatibility with all objects.